modules/repofiles/update.go

```diff
@@ -455,12 +455,58 @@ type PushUpdateOptions struct {
 	Branch       string
 }
 
+func (opts PushUpdateOptions) isNewRef() bool {
```

Comment:
can we move https://github.com/go-gitea/gitea/pull/10105/files#diff-82f7da3d2fe2bbdd9efe0e6103278d26R458-R505 to a new file. perhaps modules/repofiles/checks.go or so?